plugins/packages/appwrite/lib/operations.ts

```diff
@@ -1,87 +1,164 @@
+    if (!order_fields) throw new Error('Order fields field is required.');
+    if (!order_types) throw new Error('Order types field is required.');
+    order_fields = parseValue(order_fields);
+    order_types = parseValue(order_types);
+    if (typeof order_fields === 'string' || typeof order_types === 'string') {
+      throw new Error('Stringify order fields & order types values if not passing as {{["VALUE"]}}');
+    }
+    if (order_types.length !== order_fields.length) {
+      throw new Error('Size of order types & order fields should be same');
+    }
+
+    for (let loop = 0; loop < order_fields.length; loop++) {
+      if (order_types[loop].toLowerCase() === 'asc') {
+        queries.push(Query.orderAsc(order_fields[loop]));
+      }
+      if (order_types[loop].toLowerCase() === 'desc') {
+        queries.push(Query.orderDesc(order_fields[loop]));
+      }
+    }
+  }
+
+  return await db.listDocuments(databaseId, collection, queries);
 }
 
-export async function getDocument(db: Database, collectionId: string, documentId: string): Promise<object> {
-  return await db.getDocument(collectionId, documentId);
+export async function getDocument(
+  db: Databases,
+  databaseId: string,
+  collectionId: string,
+  documentId: string
+): Promise<AwModelDocument> {
+  return await db.getDocument(databaseId, collectionId, documentId);
 }
 
-export async function createDocument(db: Database, collectionId: string, body: object): Promise<object> {
-  return await db.createDocument(collectionId, 'unique()', body);
+export async function createDocument(
+  db: Databases,
+  databaseId: string,
+  collectionId: string,
+  body: object
+): Promise<AwModelDocument> {
+  return await db.createDocument(databaseId, collectionId, 'unique()', body);
 }
 
 export async function updateDocument(
-  db: Database,
+  db: Databases,
+  databaseId: string,
   collectionId: string,
   documentId: string,
   body: object
-): Promise<object> {
-  return await db.updateDocument(collectionId, documentId, body);
+): Promise<AwModelDocument> {
+  return await db.updateDocument(databaseId, collectionId, documentId, body);
 }
 
-export async function deleteDocument(db: Database, collectionId: string, documentId: string): Promise<object> {
-  return await db.deleteDocument(collectionId, documentId);
+export async function deleteDocument(
+  db: Databases,
+  databaseId: string,
+  collectionId: string,
+  documentId: string
+): Promise<object> {
+  await db.deleteDocument(databaseId, collectionId, documentId);
+  return { deleted: true };
 }
 
 export async function bulkUpdate(
-  db: Database,
+  db: Databases,
+  databaseId: string,
   collectionId: string,
   records: Array<object>,
   documentIdKey: string
 ): Promise<object> {
   for (const record of records) {
     const documentId = record[documentIdKey];
-    await db.updateDocument(collectionId, documentId, record);
+    await db.updateDocument(databaseId, collectionId, documentId, record);
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/ToolJet/ToolJet/pull/10047#issuecomment-2161486769>#37201c</a><span> - 06/11/2024, 07:45 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The 'await' keyword is used inside a loop. This can lead to performance issues due to sequential execution of asynchronous operations.
</div>
<div id="fix">
<b>Fix:</b> Use 'Promise.all' to execute the asynchronous operations in parallel.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div>
<pre><code>
 @@ -159,7 +159,8 @@
    for (const record of records) {
 -    const documentId = record[documentIdKey];
 -    await db.updateDocument(databaseId, collectionId, documentId, record);
 +    const updatePromises = records.map(record => {
 +      const documentId = record[documentIdKey];
 +      return db.updateDocument(databaseId, collectionId, documentId, record);
 +    });
 +    await Promise.all(updatePromises);
    }
</code></pre>
</div>
</details>
</div>
<br>


</div>